csp/baselib.py

```diff
@@ -759,32 +880,101 @@ def comp(x: ts["T"], y: ts["U"]):
     return comp
 
 
-# Math operators
+@lru_cache(maxsize=512)
+def define_unary_op(name, op_lambda):
+    float_out_type, int_out_type, generic_out_type = [None] * 3
+    if name in COMP_OPS:
+        float_out_type = bool
+        int_out_type = bool
+        generic_out_type = bool
+    elif name in MATH_OPS:
+        float_out_type = float
+        if name in ("abs",):
+            int_out_type = int
+            generic_out_type = "T"
```

Comment:
abs on a non-numeric type??

Reply:
anything that implements `__abs__` will support `abs()`.